src/site/topic23.rst

```diff
@@ -189,6 +189,64 @@ Add
 Min & Remove Min
 ----------------
 
+* Given the nature of the binary search tree ordering, the node with the minimum element **must** be in the leftmost node in the tree
+* There are three cases we'll have to take care of when looking for the minimum node
+    1. The leftmost node is the root (root has no left child)
+    2. The leftmost is a leaf node (no children)
+    3. The leftmost node is an interior node
+
+* Finding the minimum is simple
+    * Assuming a root node exists, keep going to the left subtree until there is no more left subtree
+
+* Removing will be a little trickier since we may be removing something that has children that must remain in the tree
+* And further, not only must the children stay in the tree, but the proper binary search tree ordering **must** be preserved
+
+* Consider the following examples and think of how one would need to manage potential children of a node being removed
+    * For this example, assume duplicate values will go to the right
+
+.. image:: img/binarysearchtree_remove0.png
+   :width: 500 px
+   :align: center
+
+.. image:: img/binarysearchtree_remove1.png
+   :width: 500 px
+   :align: center
+
+.. image:: img/binarysearchtree_remove2.png
+   :width: 500 px
+   :align: center
+
+* If we have case 1, where the minimum value is in the root node, this means that
+    * There is no left subtree
+    * There *may* be a right subtree
+    * Therefore, we simply make the root of the right subtree (which may be null) the new root
+
+* For case 1, the ordering is preserved since
+    * Nothing is to the left of the root
+    * Everything to the right of the root is larger than (or equal to) the value in the root
+    * Everything to the right of the right subtree's root must be larger than (or equal to) the right subtree's root
+    * Everything to the left of the right subtree's root must be less than the right subtree's root
+    * Therefore, if all subtrees start with the proper ordering, if the right subtree becomes the new root, the binary search tree's order will be maintained
```

Comment:
This is a little verbose. You can exploit the recursive definition of the binary search tree to simply the explanation (for case 3 as well). Any subtree of a binary search tree must itself be a binary search tree (it's not _if_ like the last point here says). For case 1, the tree is essentially replaced by the right subtree; by definition, order is maintained.